test/support/proposal_context.rb

```diff
@@ -67,6 +72,15 @@
     settings
   end
 
+  let(:ng_settings) do
+    file = File.join(DATA_PATH, "control_files", control_file)
+    settings = Yast::XML.XMLToYCPFile(file)
+    Yast::ProductFeatures.Import(settings)
```

Comment:
Why do you need to import them? Is not enough with returning and doing something like:

```ruby
allow(Y2Storage::ProposalSettings).to receive(:new_for_current_product)
  .and_return(settings)
```

Reply:
I don't think so because that `settings` is only a hash, and we need a `ProposalSettings` object with all parsers done over `settings` value.

Reply:
What looks wrong to me is that the `let` block has side-effects. IMHO the import should take care on a different place (in a before block, for instance).

On the other hand, I can live with that.